### PR TITLE
Fix round change

### PR DESCRIPTION
**Pull request by aneequesafdar**

### CODE REVIEW

**Comment by JekaMas** [comment]
@aneequesafdar Are you sure that we don't need to have a timeout for the waiting a proposer? AFAIK we stuck if any proposer is down and it's the round when we're waiting it's proposed block.

**Comment by aneequesafdar** [comment]
I am not sure what you mean, do you mean to have for a proposer who sends the proposal and is waiting for prevotes?

**Comment by aneequesafdar** [comment]
please refer to `fix-round-change-#156-debug` for debug messages

File: local_test_network/docker-compose.yml [hunk 1]
```diff
@@ -76,25 +76,6 @@ services:
     networks:
       chainnet:
         ipv4_address: 172.25.0.14
-  node5:
```
Comment (JekaMas) [inline local_test_network/docker-compose.yml]: Why not keep a cluster of 5 nodes?

Comment (aneequesafdar) [reply local_test_network/docker-compose.yml]: Not a good reason but the simplest case is for 4 nodes. Would you like to revert?

Comment (aneequesafdar) [reply local_test_network/docker-compose.yml]: Also, on slower machines messages are passed around more quickly, I can do that locally and not push.

Comment (aneequesafdar) [reply local_test_network/docker-compose.yml]: reverted

**Pull request merged**